transcom/mymove
# INT B-21073 add re_countries table and refactor

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-21073)

## Summary

For the last twenty years I have been working on this BL item and its beautiful personality and quirks. After fighting a timeout issue likely due to so many concurrency issues happening when building addresses (and also adding countries to the db), I updated the `db-truncate` file to NOT clear the test db of `re_countries` data so we can retain that and avoid unnecessary inserts into the test database when building test data.

OKAY.

So the point of this BL item was to add a single source of truth for country information, which required a `re_countries` table with populated data. 

In addition to adding that data, we had to then refactor how we are handling country data in the backend and adding the `Country` object to the struct and then making that data accessible to the UI.

This PR does the following:
- adds migration file that adds the `re_countries` table and refactors the `addresses` table to use the `id` column of the `re_countries` in the new `country_id` column - drops the `country` column
- updates relevant Go structs
- adds/updates a buncha `Eager`/`EagerPreload` instances to use `"Addresses.Country"` now that the `Country` object holds the necessary data
- updates the `Address` definition for Swagger to only be two characters (we will handle this later on when international rolls out)
- defaults all created addresses to have a `US` country
- updates all tests (roughly 3 million tests)

### How to test

1. Run `make server_run` and see the migrations run
2. Check your database and you should now see a `re_countries` table, as well as the `addresses` table having a removed `country` column and added `country_id` column
3. Open up the application and console log some addresses
4. You should be able to see the `Country` value in the addresses object (this is not nested at the moment, not sure if it ever will be - handling this return in the model to payload conversion)
5. Check any situation/process where `Country` is important
